src/tensor_conversion.cpp

```diff
@@ -462,49 +462,69 @@ static Status convertNativeFileFormatRequestTensorToOVTensor(const TensorType& s
         return StatusCode::NOT_IMPLEMENTED;
     }
+    int batchSize = getBinaryInputsSize(src);
     size_t maxStringLength = 0;
-    for (const auto& str : src.contents().bytes_contents()) {
-        maxStringLength = std::max(maxStringLength, str.size());
+    for (int i = 0; i < batchSize; i++) {
+        maxStringLength = std::max(maxStringLength, getBinaryInput(src, i).size());
     }
     size_t width = maxStringLength + 1;
-    tensor = ov::Tensor(ov::element::Type_t::u8, ov::Shape{static_cast<uint64_t>(src.contents().bytes_contents_size()), width});
-
-    size_t i = 0;
-    for (const auto& str : src.contents().bytes_contents()) {
-        std::memcpy(tensor.data<unsigned char>() + i * width, reinterpret_cast<const unsigned char*>(str.c_str()), str.size());
-        tensor.data<unsigned char>()[i * width + str.size()] = 0;
-        i++;
+    tensor = ov::Tensor(ov::element::Type_t::u8, ov::Shape{static_cast<uint64_t>(batchSize), width});
+    for (int i = 0; i < batchSize; i++) {
+        std::memcpy(
+            tensor.data<unsigned char>() + i * width,
+            reinterpret_cast<const unsigned char*>(getBinaryInput(src, i).c_str()),
+            getBinaryInput(src, i).size());
+        for (size_t j = getBinaryInput(src, i).size(); j < width; j++) {
+            tensor.data<unsigned char>()[i * width + j] = 0;
+        }
     }
     return StatusCode::OK;
 }
 
-Status convertStringRequestTensorToOVTensor(const tensorflow::TensorProto& src, ov::Tensor& tensor, const std::string* buffer) {
-    OVMS_PROFILE_FUNCTION();
-    size_t maxStringLength = 0;
-    for (const auto& str : src.string_val()) {
-        maxStringLength = std::max(maxStringLength, str.size());
-    }
-    size_t width = maxStringLength + 1;
-    tensor = ov::Tensor(ov::element::Type_t::u8, ov::Shape{static_cast<uint64_t>(src.string_val_size()), width});
-    size_t i = 0;
-    for (const auto& str : src.string_val()) {
-        std::memcpy(tensor.data<unsigned char>() + i * width, reinterpret_cast<const unsigned char*>(str.c_str()), str.size());
-        tensor.data<unsigned char>()[i * width + str.size()] = 0;
-        i++;
+template <typename TensorType>
+Status convertStringRequesto1DOVTensor(const TensorType& src, ov::Tensor& tensor, const std::string* buffer) {
+    int batchSize = getBinaryInputsSize(src);
+    int64_t totalStringsLength = 0;
+    for (int i = 0; i < batchSize; i++) {
+        totalStringsLength += getBinaryInput(src, i).size();
+    }
+    int64_t width = totalStringsLength + sizeof(uint32_t) * (batchSize + 2);
+    tensor = ov::Tensor(ov::element::Type_t::u8, ov::Shape{static_cast<size_t>(width)});
+    uint32_t* data = reinterpret_cast<uint32_t*>(tensor.data<uint8_t>());
+    data[0] = static_cast<uint32_t>(batchSize);
+    data[1] = 0;  // first string start offset
+    unsigned char* condensedContentStart = tensor.data<unsigned char>() + sizeof(uint32_t) * (batchSize + 2);
```

Comment:
Maybe we can find meaningful name for batchSize + 2 value. Also comment with explanation what is going on in this function would be helpful.

Reply:
done